### ISSUE TITLE
Feature Request: `AWS::StepFunctions::StateMachine` - (JSONPath) Recognize that `OutputPath` needs to start with "$"

### ISSUE THREAD

**Issue by r-heimann** [aws-cloudformation/cfn-lint#4078]
### Is this feature request related to a new rule or cfn-lint capabilities?

New capability

### Describe the feature you'd like to request

Currently cfn-lint is unable to recognize that `OutputPath` needs to starts with "$". The following CloudFormation Template will cause a CloudFormation error:

```yaml
AWSTemplateFormatVersion: 2010-09-09
Description: Test

Resources:
  StepFunctionRole:
    Type: AWS::IAM::Role
    Properties:
      AssumeRolePolicyDocument:
        Statement:
          - Effect: Allow
            Principal:
              Service: states.amazonaws.com
            Action: sts:AssumeRole

  StateMachine:
    Type: AWS::StepFunctions::StateMachine
    Properties:
      RoleArn: !GetAtt StepFunctionRole.Arn
      Definition:
        StartAt: Pass
        States:
          Pass:
            Type: Pass
            OutputPath: "Test" # <---------------- needs to start with $
            Next: Success
          Success:
            Type: Succeed
```

CloudFormation error:

```
Resource handler returned message:
"Invalid State Machine Definition: 'SCHEMA_VALIDATION_FAILED: Value must be a valid JSONPath. at /States/Pass/OutputPath'
(Service: Sfn, Status Code: 400, Request ID: 5...3)
(SDK Attempt Count: 1)"
(RequestToken: 9...4, HandlerErrorCode: InvalidRequest)
```

It would be great if cfn-lint could give an error on this.

### Describe the solution you'd like

A new/updated rule to notify the user of this error.

### Is this something that you'd be interested in working on?

- [ ] 👋 I may be able to implement this feature request

### Would this feature include a breaking change?

- [ ] ⚠️ This feature might incur a breaking change

**Comment by r-heimann** [comment]
Related to https://github.com/aws-cloudformation/cfn-lint/issues/4076

Also affects

- InputPath
- ResultPath
- Parameters/<Name>
- Result/<Name>
- Assign/<Name>
